### PR TITLE
refactor: port to modern hooks

### PR DESCRIPTION
**Pull request by LordMidas**

### CODE REVIEW

File: scripts/!mods_preload/mod_reforged.nut [hunk 1]
```diff
@@ -3,11 +3,31 @@
...
+	"mod_dynamic_spawns",
+	"mod_item_tables",
+	"mod_upd",
+	"mod_stack_based_skills"
+);
+::Reforged.HooksMod.conflictWith(
+	"mod_legends"
+);
+
+::Reforged.HooksMod.queue(function() {
```
Comment (LordMidas) [inline scripts/!mods_preload/mod_reforged.nut:30]: Add manual queue ordering

File: scripts/skills/actives/rf_between_the_eyes_skill.nut [hunk 2]
```diff
@@ -94,12 +94,12 @@ this.rf_between_the_eyes_skill <- ::inherit("scripts/skills/skill", {
 		if (aoo != null)
 		{
 			this.m.DamageType = aoo.getDamageType().weakref();
-			this.m.ActionPointCost += aoo.m.ActionPointCost;
-			this.m.FatigueCost += aoo.m.FatigueCost;
-			this.m.FatigueCostMult = aoo.m.FatigueCostMult;
-			this.m.DirectDamageMult = aoo.m.DirectDamageMult;
-			this.m.MinRange = aoo.m.MinRange;
-			this.m.MaxRange = aoo.m.MaxRange;
+			this.m.ActionPointCost += aoq.m.ActionPointCost;
```
Comment (LordMidas) [inline scripts/skills/actives/rf_between_the_eyes_skill.nut:97]: aoo.m.

File: scripts/skills/actives/rf_between_the_eyes_skill.nut [hunk 3]
```diff
@@ -140,14 +140,14 @@ this.rf_between_the_eyes_skill <- ::inherit("scripts/skills/skill", {
 	{
 		local aoo = this.getContainer().getAttackOfOpportunity()
 
-		local overlay = aoo.m.Overlay;
-		aoo.m.Overlay = "";
+		local overlay = aoq.m.Overlay;
```
Comment (LordMidas) [inline scripts/skills/actives/rf_between_the_eyes_skill.nut:143]: aoo.m.

File: scripts/skills/actives/rf_swordmaster_kick_skill.nut [hunk 4]
```diff
@@ -131,10 +131,10 @@ this.rf_swordmaster_kick_skill <- ::inherit("scripts/skills/actives/rf_swordmast
 		}
 
 		local aoo = this.getContainer().getAttackOfOpportunity();
-		local overlay = aoo.m.Overlay;
-		aoo.m.Overlay = "";
+		local overlay = aoq.m.Overlay;
```
Comment (LordMidas) [inline scripts/skills/actives/rf_swordmaster_kick_skill.nut:134]: aoo.m.

File: scripts/skills/actives/rf_swordmaster_push_through_skill.nut [hunk 5]
```diff
@@ -87,10 +87,10 @@ this.rf_swordmaster_push_through_skill <- ::inherit("scripts/skills/actives/line
 		}
 
 		local aoo = this.getContainer().getAttackOfOpportunity();
-		local overlay = aoo.m.Overlay;
-		aoo.m.Overlay = "";
+		local overlay = aoq.m.Overlay;
```
Comment (LordMidas) [inline scripts/skills/actives/rf_swordmaster_push_through_skill.nut:90]: aoo.m.

File: scripts/skills/actives/rf_swordmaster_tackle_skill.nut [hunk 6]
```diff
@@ -118,10 +118,10 @@ this.rf_swordmaster_tackle_skill <- ::inherit("scripts/skills/actives/rf_swordma
 		}
 
 		local aoo = this.getContainer().getAttackOfOpportunity();
-		local overlay = aoo.m.Overlay;
-		aoo.m.Overlay = "";
+		local overlay = aoq.m.Overlay;
```
Comment (LordMidas) [inline scripts/skills/actives/rf_swordmaster_tackle_skill.nut:121]: aoo.m.

File: mod_reforged/hooks/entity/tactical/actor.nut [hunk 7]
```diff
@@ -91,16 +88,15 @@
 		if (verifySettingValue("TacticalTooltip_Effects")) ret.extend(::Reforged.TacticalTooltip.getTooltipEffects(this, 200));
 		if (verifySettingValue("TacticalTooltip_Perks")) ret.extend(::Reforged.TacticalTooltip.getTooltipPerks(this, 300));
 		if (verifySettingValue("TacticalTooltip_EquippedItems")) ret.extend(::Reforged.TacticalTooltip.getTooltipEquippedItems(this, 400));
-		if (verifySettingValue("TacticalTooltip_BagItems")) ret.extend(::Reforged.TacticalTooltip.getTooltipBagItems(this, 400));
+		if (verifySettingValue("TacticalTooltip_BagItems")) ret.extend(::Reforged.TacticalTooltip.__originalBagItems(this, 400));
```
Comment (Darxo) [inline mod_reforged/hooks/entity/tactical/actor.nut:91]: This replacement seems wrong and by accident

File: scripts/!mods_preload/mod_reforged.nut [hunk 8]
```diff
@@ -50,3 +70,11 @@
 		::include(file);
 	}
 });
+
+::Reforged.HooksMod.queue(">mod_msu", ">mod_dynamic_perks", ">mod_dynamic_spawns", ">mod_item_tables", ">mod_upd", ">mod_stack_based_skills",function() {
```
Comment (Darxo) [inline scripts/!mods_preload/mod_reforged.nut:74]: missing a space between `,` and `function`

**Review by Darxo — CHANGES_REQUESTED** [review]
I looked over every single file. Most of them very briefly.

There is just one problem that will likely cause an issue during playtesting.

**Review by Darxo — APPROVED** [review]

**Pull request merged**